src/hooks/useLocaleFormatter/useLocaleFormatter.ts

```diff
@@ -122,8 +124,13 @@ type useLocaleFormatterArgs = {
  */
 
 export const useLocaleFormatter = (args?: useLocaleFormatterArgs): NumberFormatter => {
-  const currencyFormat = useAppSelector(selectCurrencyFormat)
-  const deviceLocale = args?.locale ?? currencyFormat
+  let currencyFormat = useAppSelector(selectCurrencyFormat) as string
+  // Get the default browser locale if no specific locale has been set for the currency format
+  const deviceLocale =
+    args?.locale ??
+    (currencyFormat === CurrencyFormats.SystemDefault
+      ? defaultBrowserCurrencyFormat()
+      : currencyFormat)
```

Comment:
Abstract to `currencyFormatOrSystemDefault` ?

Reply:
Abstracted!